backend/dataline/services/result.py

```diff
@@ -118,3 +115,37 @@ async def validate_sql_query_result_for_chart(
                 "New SQL query is not compatible with chart! "
                 "Make sure to specify 2 columns, first for labels and second for values."
             )
+
+    async def export_results_as_csv(self, session: AsyncSession, result_id: UUID) -> StreamingResponse:
+        # Fetch the SQL_QUERY_STRING_RESULT
+        query_string_result = await self.result_repo.get_by_uuid(session, result_id)
+        if query_string_result.type != QueryResultType.SQL_QUERY_STRING_RESULT.value:
+            raise ValueError("The provided result_id does not belong to an SQL_QUERY_STRING_RESULT")
+
+        # Parse the SQL query from the content
+        query_content = SQLQueryStringResultContent.model_validate_json(query_string_result.content)
+        sql_query = query_content.sql
+
+        # Get the connection for the result
+        connection = await self.result_repo.get_connection_from_result(session, result_id)
+        db = SQLDatabase.from_dataline_connection(Connection.model_validate(connection))
+
+        # Create a generator function to stream the results
+        async def generate_csv():
```

Comment:
I would extract this into a separate function that takes as input a generator and creates a csv generator from it.

Not a fan of in-line function definitions, unless it really makes sense. I would still separate it even if this is only used in a single place. Readability counts more than most things imo